modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/ArtifactStoppedDeploymentPersistenceListener.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.deployment.internal;
+
+import static java.lang.String.valueOf;
+import static java.util.Optional.of;
+import static org.mule.runtime.module.deployment.impl.internal.util.DeploymentPropertiesUtils.resolveDeploymentProperties;
+import static org.mule.runtime.module.deployment.internal.DefaultArchiveDeployer.START_ARTIFACT_ON_DEPLOYMENT_PROPERTY;
+
+import org.mule.runtime.core.internal.context.ArtifactStoppedPersistenceListener;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class ArtifactStoppedDeploymentPersistenceListener implements ArtifactStoppedPersistenceListener {
+
+  private transient final Logger logger = LoggerFactory.getLogger(getClass());
+  private AtomicBoolean shouldPersist;
+  private String artifactName;
+
+  public ArtifactStoppedDeploymentPersistenceListener(String artifactName) {
+    this.artifactName = artifactName;
+    shouldPersist = new AtomicBoolean(true);
```

Comment:
where can this be used concurrently?

Reply:
We thought that a possible race condition could happen if a stop request and a shutdown request were concurrently sent to mule, so we decided to make this variable an Atomic Boolean.

Reply:
add this info as a comment in the code